ipa-core/src/helpers/gateway/mod.rs

```diff
@@ -84,6 +85,10 @@ pub struct GatewayConfig {
     /// payload may not be exactly this, but it will be the closest multiple of record size to this
     /// number. For instance, having 14 bytes records and batch size of 4096 will result in
     /// 4088 bytes being sent in a batch.
+    ///
+    /// The actual size for read chunks may be bigger or smaller, depending on the record size
+    /// sent through each channel. Read size will be aligned with [`Self::active_work`] value to
```

Comment:
Add a comment explaining what to do if we want to align read size perfectly with the target read size

Reply:
I suggest rewriting this entire comment. It feels out of date.

- line 85 has a typo "side" instead of "size".
- line 86 refers to "batch size" but this comment is above "read_size".
- I think the example is wrong, because 4088 bytes is not a power of two multiple of 14. If I'm not wrong, the best you can do is now 14 * 2^8 < 4096 (it's 3,584).

Reply:
You are absolutely right. Just did that